src/System.Runtime.Extensions/tests/System/Environment.SetEnvironmentVariable.cs

```diff
@@ -14,9 +14,9 @@ public class SetEnvironmentVariable
         private const int MAX_VAR_LENGTH_ALLOWED = 32767;
         private const string NullString = "\u0000";
 
-        private static bool IsSupportedTarget(EnvironmentVariableTarget target)
+        internal static bool IsSupportedTarget(EnvironmentVariableTarget target)
         {
-            return target == EnvironmentVariableTarget.Process || RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+            return target == EnvironmentVariableTarget.Process || (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && !PlatformDetection.IsWinRT && !PlatformDetection.IsNetNative);
```

Comment:
Incidentally, `IsWinRT` is actually testing `IsAModernApp` and ought to be renamed to something like that. Right now that means it's firing only for our F5 runs. We ought to also add something like `PlatformDetection.IsInAppcontainer` which is what most people want when they use `IsWinRT`.